packages/lib/src/project/AppProject.ts

```diff
@@ -0,0 +1,154 @@
+  public static async fromSimpleProject(simpleProject: SimpleProject, version: string = DEFAULT_VERSION, existingAddresses: any = {}): Promise<AppProject> {
+    const appProject: AppProject = await this.fetchOrDeploy(simpleProject.name, version, simpleProject.txParams, existingAddresses);
+
+    await Promise.all(
+      _.concat(
+        _.map(simpleProject.implementations, (contractInfo, contractAlias) => (
+          appProject.registerImplementation(contractAlias, contractInfo)
+        )),
+        _.map(simpleProject.dependencies, (dependencyInfo, dependencyName) => (
+          appProject.setDependency(dependencyName, dependencyInfo.package, dependencyInfo.version)
+        ))
+      ));
+    return appProject;
+  }
+
+  constructor(app: App, name: string = DEFAULT_NAME, version: string = DEFAULT_VERSION, txParams: any = {}) {
+    super(txParams);
+    this.app = app;
+    this.name = name;
+    this.version = semanticVersionToString(version);
+  }
+
+  public async newVersion(version: any): Promise<ImplementationDirectory> {
+    version = semanticVersionToString(version);
+    const directory: ImplementationDirectory = await super.newVersion(version);
+    const thepackage: Package = await this.getProjectPackage();
+    await this.app.setPackage(this.name, thepackage.address, version);
+    return directory;
+  }
+
+  public getApp(): App {
+    return this.app;
+  }
+
+  public async getProjectPackage(): Promise<Package> {
+    if (!this.package) {
+      const packageInfo: { package: Package, version: any } = await this.app.getPackage(this.name);
+      this.package = packageInfo.package;
+    }
+    return this.package;
+  }
+
+  public async getCurrentDirectory(): Promise<ImplementationDirectory> {
+    if (!this.directory) {
+      this.directory = await this.app.getProvider(this.name);
+    }
+    return this.directory;
+  }
+
+  public async getCurrentVersion(): Promise<string> {
+    return this.version;
+  }
+
+  // TODO: Testme
+  public async getImplementation({ packageName, contractName }: { contractName: string, packageName?: string }): Promise<string> {
+    return this.app.getImplementation(packageName || this.name, contractName);
+  }
+
+  // TODO: Testme
+  public async createContract(contractClass: ContractFactory, { packageName, contractName, initMethod, initArgs }: { packageName?: string, contractName?: string, initMethod?: string, initArgs?: string[] } = {}): Promise<ContractWrapper> {
+    if (!contractName) contractName = contractClass.contractName;
+    if (!packageName) packageName = this.name;
+    return this.app.createContract(contractClass, packageName, contractName, initMethod, initArgs);
+  }
+
+  public async createProxy(contractClass: ContractFactory, { packageName, contractName, initMethod, initArgs }: { packageName?: string, contractName?: string, initMethod?: string, initArgs?: string[] } = {}): Promise<ContractWrapper> {
```

Comment:
As comment above, but for the object with the properties: packageName, contractName, etc.

Reply:
They are defined. But the definition of variable types inside an object is a bit annoying and confusing
```javascript
{ packageName, contractName, initMethod, initArgs }: { packageName?: string, contractName?: string, initMethod?: string, initArgs?: string[] }
```